compiler/rustc_const_eval/src/interpret/validity.rs

```diff
@@ -1274,11 +1274,11 @@ impl<'rt, 'tcx, M: Machine<'tcx>> ValueVisitor<'tcx, M> for ValidityVisitor<'rt,
         // FIXME: We could avoid some redundant checks here. For newtypes wrapping
         // scalars, we do the same check on every "level" (e.g., first we check
         // MyNewtype and then the scalar in there).
+        if val.layout.is_uninhabited() {
```

Comment:
Nit: this is now in a section titled "check the ABI", which doesn't make a lot of sense.

Reply:
I pushed a commit addressing this.